tutorials/3d/variable_rate_shading.rst

```diff
@@ -0,0 +1,116 @@
+.. _doc_variable_rate_shading:
+
+Variable rate shading
+=====================
+
+What is variable rate shading?
+------------------------------
+
+In modern 3D rendering engines, shaders are much more complex compared to before.
+The advent of physically-based rendering, real-time global illumination and
+screen-space effects has increased the number of *per-pixel* shading that must
+be performed to render each frame. Additionally, screen resolutions also have
+increased a lot, with 1440p and 4K now being common.
+
+As a result, the total shading cost in scene rendering usually represents a
+significant amount of the time taken to render each frame.
+
+Variable rate shading (VRS) is a method of decreasing this shading cost by
+reducing the resolution of *per-pixel* shading (also called *fragment* shading),
+while keeping the original resolution for rendering geometry. VRS allows
+specifying the shading quality in a local manner, which makes it possible to
+have certain parts of the viewport receive more detailed shading than others.
+This is particularly useful in virtual reality (VR) to achieve *foveated
+rendering*, where the center of the viewport is more detailed than the edges.
+
+Hardware support
+----------------
+
+Variable rate shading is only supported on specific desktop and laptop graphics cards:
+
+- NVIDIA Turing and newer (including GTX 1600 series)
+- AMD RDNA2 and newer (both integrated and dedicated GPUs – including Steam Deck)
+- Intel Arc Alchemist and newer **(dedicated GPUs only)**
+
+  - Intel integrated graphics do not support variable rate shading.
+
+As of November 2022, mobile GPUs do not support variable rate shading.
```

Comment:
Is this correct? I thought that was one of the important things about the GPUs in Meta Quests, they definitely utilize VRS but maybe they do their own thing here...

Reply:
Indeed, it looks like my statement was wrong. The Snapdragon 888 and newer support variable rate shading: https://www.androidcentral.com/qualcomm-snapdragon-888

Here's a benchmark: https://hothardware.com/news/heres-how-current-phones-perform-variable-rate-shading-games